test/Microsoft.IdentityModel.TestUtils/IdentityComparer.cs

```diff
@@ -1669,5 +1669,14 @@ private static string GetString(object str)
             else
                 return string.Format(CultureInfo.InvariantCulture, "{0}", (str ?? "null"));
         }
+
+        public static bool AreDatesEqualWithEpsilon(DateTime? dateTime1, DateTime? dateTime2, int epsilon)
+        {
+            if (dateTime1 is DateTime date1 && dateTime2 is DateTime date2)
```

Comment:
nit: you can do HasValue instead

Reply:
Would it remove the optionality though? The resulting TimeSpan from substracting two optional DateTimes results in an optional as well, and I cannot call TotalSeconds on it.

Reply:
dateTime1 to date1 already remove the optionality doesn't it?